services/src/handlers/datasets.rs

```diff
@@ -97,8 +139,189 @@ fn adjust_user_path_to_upload_path(meta: &mut MetaDataDefinition, upload: &Uploa
+            time: OgrSourceDatasetTimeType::None, // TODO: auto detect time type and corresponding columns
+            columns: Some(OgrSourceColumnSpec {
+                x: "".to_owned(), // TODO: for csv-files: try to find wkt/xy columns
+                y: None,
+                numeric: columns_vecs.numeric,
+                decimal: columns_vecs.decimal,
+                textual: columns_vecs.textual,
+            }),
+            default_geometry: None,
+            force_ogr_time_filter: false,
+            on_error: geoengine_operators::source::OgrSourceErrorSpec::Skip,
+            provenance: None,
+        },
+        result_descriptor: VectorResultDescriptor {
+            data_type: vector_type,
+            spatial_reference: spatial_reference.into(),
+            columns: columns_map,
+        },
+    }))
+}
+
+fn detect_vector_type(layer: &Layer) -> Result<VectorDataType> {
+    let ogr_type = layer
+        .feature(0)
+        .ok_or(error::Error::EmptyDataSetCannotBeImported)?
+        .geometry()
+        .geometry_type();
+
+    VectorDataType::try_from_ogr_type_code(ogr_type).context(error::DataType)
+}
+
+struct Columns {
+    decimal: Vec<String>,
+    numeric: Vec<String>,
+    textual: Vec<String>,
+}
+
+fn detect_columns(layer: &Layer) -> HashMap<String, FeatureDataType> {
+    let mut columns = HashMap::default();
+    // TODO: avoid reading ALL of the data
+    for feature in layer.features() {
+        for (name, value) in feature.fields() {
+            #[allow(clippy::match_same_arms)]
+            match value {
+                gdal::vector::FieldValue::IntegerValue(_)
+                | gdal::vector::FieldValue::Integer64Value(_) => {
+                    // only store as integer if there is no other value type present in the column
+                    #[allow(clippy::map_entry)]
+                    if !columns.contains_key(&name) {
+                        columns.insert(name, FeatureDataType::Decimal);
+                    }
+                }
+                gdal::vector::FieldValue::RealValue(_) => {
+                    if let Some(column) = columns.get(&name) {
+                        if *column == FeatureDataType::Decimal {
+                            // migrate integer column to float
+                            columns.insert(name, FeatureDataType::Number);
+                        }
+                    } else {
+                        columns.insert(name, FeatureDataType::Number);
+                    }
+                }
+                gdal::vector::FieldValue::StringValue(_) => {
+                    columns.insert(name, FeatureDataType::Text);
+                }
+                gdal::vector::FieldValue::DateValue(_)
+                | gdal::vector::FieldValue::DateTimeValue(_) => {
+                    // TODO: ensure all feature's field by this name are a date/datetime
+                    // TODO: auto detect if this is start/end date/datetime
```

Comment:
I guess we don't. We should just assume it and then we have our error handling, for instance, set null.

Reply:
c8e6fe123441d3cfe993329a1edf7663afe85fbd